osu.Game/Screens/Play/HUD/FailingLayer.cs

```diff
@@ -74,7 +75,8 @@ private void load(OsuColour color, OsuConfigManager config)
             boxes.Colour = color.Red;
 
             configEnabled = config.GetBindable<bool>(OsuSetting.FadePlayfieldWhenHealthLow);
-            enabled.BindValueChanged(e => this.FadeTo(e.NewValue ? 1 : 0, fade_time, Easing.OutQuint), true);
+            enabled.BindValueChanged(e => TryToFade(fade_time, Easing.OutQuint, e.NewValue), true);
+            HUDEnabled.BindValueChanged(e => TryToFade(fade_time, Easing.OutQuint, e.NewValue), true);
```

Comment:
Why introduce another bindable, that only convoluted logic, shouldn't you only hide the layer in the `HUDOverlay.ShowHealthbar` bindable changes, and add an internal `Container` layer to hide the layer when `enabled` is false.

Reply:
@bdach Are you sure about having the visibility logic change internally here, I think it shouldn't.

Reply:
I'd rather have it all in one place myself, rather than half in `Player` and half here.

Reply:
So do I keep it as it is or not?

Reply:
@bdach Fair enough. @power9maker No, remove it and just put a check inside `enabled.BindValueChanged(...)` to control the visibility based on "show health display ..." setting and new value of `enabled`.

Reply:
Well yes, but I would have to add `TryToFade()` to `HUDOverlay`.
Is that what you want?